src/internal/gateway/gateway.go

```diff
@@ -179,7 +180,10 @@ func (g *Gateway) listenAndServe() {
 }
 
 func (g *Gateway) handleInfoEndpoint(w http.ResponseWriter, r *http.Request) {
-	w.Write([]byte(fmt.Sprintf(`{"version":"%s","vm_uptime":"%d"}`+"\n", g.logCacheVersion, g.uptimeFn())))
+	_, err := w.Write([]byte(fmt.Sprintf(`{"version":"%s","vm_uptime":"%d"}`+"\n", g.logCacheVersion, g.uptimeFn())))
+	if err != nil {
+		g.log.Fatalf("Cannot send result for the info endpoint")
```

Comment:
Why should this be a fatal log when it wasn't before? Perhaps this should just be an error log.